{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "SinGAN.ipynb",
      "provenance": [],
      "authorship_tag": "ABX9TyMBI0OL42WlgpJPzirjLBXg",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    },
    "accelerator": "GPU"
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/nulllpointer/Thesis/blob/master/SinGAN.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from fastai.vision import *\n",
        "from fastai.metrics import error_rate\n",
        "from PIL import ImageFile\n",
        "ImageFile.LOAD_TRUNCATED_IMAGES = True\n",
        "import warnings\n",
        "warnings.filterwarnings(\"ignore\", category=UserWarning, module=\"torch.nn.functional\")"
      ],
      "metadata": {
        "id": "xoECpxPPkQ7M"
      },
      "execution_count": 15,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "from google.colab import drive\n",
        "from pathlib import Path\n",
        "drive.mount('/content/drive')\n",
        "path= '/content/drive/MyDrive/rice'\n",
        "path = Path(path)\n",
        "original_bacterial_data_path = '/content/drive/MyDrive/rice/original_dataset/Bacterial_leaf_blight'\n",
        "aug_data_path = path/'Augmented/Bacterial_Leaf_Blight'\n",
        "aug_data_path\n"
      ],
      "metadata": {
        "id": "GT9EIBC2kRmG",
        "outputId": "c85670e1-a815-4d9b-8e1e-47bad435c170",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "execution_count": 16,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "PosixPath('/content/drive/MyDrive/rice/Augmented/Bacterial_Leaf_Blight')"
            ]
          },
          "metadata": {},
          "execution_count": 16
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# from PIL import Image\n",
        "# import glob\n",
        "# image_list = []\n",
        "# # for filename in glob.glob('original_data_path/*.jpg'): \n",
        "# for filename in glob.glob(original_bacterial_data_path+'/*.JPG'): \n",
        "#     print(filename)\n",
        "#     im=Image.open(filename)\n",
        "#     image_list.append(im)\n",
        "\n",
        "# for filename in glob.glob(original_bacterial_data_path+'/*.jpg'): \n",
        "#     print(filename)\n",
        "#     im=Image.open(filename)\n",
        "#     image_list.append(im)\n",
        "\n",
        "# len(image_list)\n"
      ],
      "metadata": {
        "id": "9aU9ByPypAQo"
      },
      "execution_count": 17,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3EgzCdjT8ij6",
        "outputId": "31d7095f-0372-4a30-a3f4-e000f9b10597"
      },
      "source": [
        "%cd /root\n",
        "!ls\n",
        "!git clone https://github.com/nulllpointer/Thesis.git\n",
        "%cd /root/Thesis/SinGAN\n",
        "!python -m pip install -r requirements.txt"
      ],
      "execution_count": 18,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/root\n",
            "Thesis\n",
            "fatal: destination path 'Thesis' already exists and is not an empty directory.\n",
            "/root/Thesis/SinGAN\n",
            "Requirement already satisfied: matplotlib in /usr/local/lib/python3.7/dist-packages (from -r requirements.txt (line 1)) (3.2.2)\n",
            "Requirement already satisfied: scikit-image in /usr/local/lib/python3.7/dist-packages (from -r requirements.txt (line 2)) (0.18.3)\n",
            "Requirement already satisfied: scikit-learn in /usr/local/lib/python3.7/dist-packages (from -r requirements.txt (line 3)) (1.0.2)\n",
            "Requirement already satisfied: scipy in /usr/local/lib/python3.7/dist-packages (from -r requirements.txt (line 4)) (1.4.1)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.7/dist-packages (from -r requirements.txt (line 5)) (1.21.6)\n",
            "Requirement already satisfied: torch==1.4.0 in /usr/local/lib/python3.7/dist-packages (from -r requirements.txt (line 6)) (1.4.0)\n",
            "Requirement already satisfied: torchvision==0.5.0 in /usr/local/lib/python3.7/dist-packages (from -r requirements.txt (line 7)) (0.5.0)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.7/dist-packages (from torchvision==0.5.0->-r requirements.txt (line 7)) (1.15.0)\n",
            "Requirement already satisfied: pillow>=4.1.1 in /usr/local/lib/python3.7/dist-packages (from torchvision==0.5.0->-r requirements.txt (line 7)) (7.1.2)\n",
            "Requirement already satisfied: python-dateutil>=2.1 in /usr/local/lib/python3.7/dist-packages (from matplotlib->-r requirements.txt (line 1)) (2.8.2)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.7/dist-packages (from matplotlib->-r requirements.txt (line 1)) (0.11.0)\n",
            "Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.7/dist-packages (from matplotlib->-r requirements.txt (line 1)) (1.4.2)\n",
            "Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 in /usr/local/lib/python3.7/dist-packages (from matplotlib->-r requirements.txt (line 1)) (3.0.8)\n",
            "Requirement already satisfied: typing-extensions in /usr/local/lib/python3.7/dist-packages (from kiwisolver>=1.0.1->matplotlib->-r requirements.txt (line 1)) (4.2.0)\n",
            "Requirement already satisfied: PyWavelets>=1.1.1 in /usr/local/lib/python3.7/dist-packages (from scikit-image->-r requirements.txt (line 2)) (1.3.0)\n",
            "Requirement already satisfied: imageio>=2.3.0 in /usr/local/lib/python3.7/dist-packages (from scikit-image->-r requirements.txt (line 2)) (2.4.1)\n",
            "Requirement already satisfied: networkx>=2.0 in /usr/local/lib/python3.7/dist-packages (from scikit-image->-r requirements.txt (line 2)) (2.6.3)\n",
            "Requirement already satisfied: tifffile>=2019.7.26 in /usr/local/lib/python3.7/dist-packages (from scikit-image->-r requirements.txt (line 2)) (2021.11.2)\n",
            "Requirement already satisfied: threadpoolctl>=2.0.0 in /usr/local/lib/python3.7/dist-packages (from scikit-learn->-r requirements.txt (line 3)) (3.1.0)\n",
            "Requirement already satisfied: joblib>=0.11 in /usr/local/lib/python3.7/dist-packages (from scikit-learn->-r requirements.txt (line 3)) (1.1.0)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "scDh9AU1_H0J",
        "outputId": "faed87e1-2476-420a-88a0-23f45a10a7a0"
      },
      "source": [
        "import os\n",
        "os.chdir('/root/Thesis/SinGAN')\n",
        "!rm -rf /root/Thesis/SinGAN/Input/Images/*\n",
        "!rm -rf /root/Thesis/SinGAN/TrainedModels/*\n",
        "!rm -rf /root/Thesis/SinGAN/Output/RandomSamples/*\n",
        "\n",
        "!cp -r /content/drive/MyDrive/rice/original_dataset/Bacterial_leaf_blight/* /root/Thesis/SinGAN/Input/Images\n",
        "\n",
        "\n",
        "input_dir_path='/root/Thesis/SinGAN/Input/Images'\n",
        "\n",
        "img_file_lst = os.listdir(input_dir_path)\n",
        "print(len(img_file_lst))\n",
        "\n",
        "from pathlib import Path\n",
        "\n",
        "for img in img_file_lst:\n",
        " \n",
        "  !python main_train.py --input_name $img\n",
        "# #   # # To generate random samples from any starting generation scale, please first train SinGAN model on the desired image (as described above), then run\n",
        "  !python random_samples.py --input_name $img --mode random_samples --gen_start_scale 0\n",
        "  !mkdir -p /tmp/$img\n",
        "  !cp -r /root/Thesis/SinGAN/Output/RandomSamples/* /tmp/$img/\n",
        "  !cp -r /tmp/$img /content/drive/MyDrive/rice/Augmented/Bacterial_Leaf_Blight\n",
        "  !rm -rf /root/Thesis/SinGAN/TrainedModels/*\n",
        "  !rm -rf /root/Thesis/SinGAN/Output/RandomSamples/*\n",
        "  "
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "40\n",
            "Random Seed:  2817\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  2010\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  9464\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  2725\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  1886\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  4527\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  2529\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  2039\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  1141\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  7579\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  2207\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  2126\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  9506\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  3742\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  1372\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  9142\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  4758\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  6772\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  3320\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  2223\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  1411\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  2064\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  1127\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  6913\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  3145\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  838\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  8889\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  4365\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  2391\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  4590\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  2994\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  1173\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  7608\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  8788\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  4384\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  7209\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  573\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  4043\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"random_samples.py\", line 38, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n",
            "Random Seed:  6063\n",
            "THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=50 error=100 : no CUDA-capable device is detected\n",
            "Traceback (most recent call last):\n",
            "  File \"main_train.py\", line 27, in <module>\n",
            "    real = functions.read_image(opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 151, in read_image\n",
            "    x = np2torch(x,opt)\n",
            "  File \"/root/Thesis/SinGAN/SinGAN/functions.py\", line 172, in np2torch\n",
            "    x = x.type(torch.cuda.FloatTensor) if not(opt.not_cuda) else x.type(torch.FloatTensor)\n",
            "RuntimeError: cuda runtime error (100) : no CUDA-capable device is detected at /pytorch/aten/src/THC/THCGeneral.cpp:50\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        ""
      ],
      "metadata": {
        "id": "Tx2j-DjWlbgQ"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        ""
      ],
      "metadata": {
        "id": "pyS3p7p1lbc4"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        ""
      ],
      "metadata": {
        "id": "RZlDquVTlbZ0"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        ""
      ],
      "metadata": {
        "id": "gYHEhYXRlbRV"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}